### PR TITLE
Registering only valid Odoo modules from custom dir

### PR DESCRIPTION
**Pull request by PCatinean**
@Yajo fix for  https://github.com/Tecnativa/docker-odoo-base/issues/91

### CODE REVIEW

**Comment by PCatinean** [comment]
Not sure what's going on with the last failing test

File: lib/odoobaselib/__init__.py [hunk 1]
```diff
@@ -57,6 +58,15 @@ def addons_config():
...
                         logging.debug("Expanding glob %s", glob)
                         for addon in iglob(os.path.join(SRC_DIR, repo, glob)):
+                            if not os.path.isdir(addon):
+                                continue
+                            addon_files = set(os.listdir(addon))
+                            has_manifest = manifest_files & addon_files
+                            py_module = '__init__.py' in addon_files
+                            if not py_module or not has_manifest:
+                                logging.debug("Skipping '%s' as it is not a "
+                                    "valid Odoo module" % addon)
```
Comment (Yajo) [inline lib/odoobaselib/__init__.py]: Please do proper indentation here 😊

File: lib/odoobaselib/__init__.py [hunk 2]
```diff
@@ -57,6 +58,15 @@ def addons_config():
                     for glob in addons:
                         logging.debug("Expanding glob %s", glob)
                         for addon in iglob(os.path.join(SRC_DIR, repo, glob)):
+                            if not os.path.isdir(addon):
+                                continue
+                            addon_files = set(os.listdir(addon))
+                            has_manifest = manifest_files & addon_files
+                            py_module = '__init__.py' in addon_files
```
Comment (Yajo) [inline lib/odoobaselib/__init__.py]: I think the `py_module` check can be misleading. What if a `setup` dir contains one of `__init__.py` files? `has_manifest` should be enough IMHO.

File: lib/odoobaselib/__init__.py [hunk 3]
```diff
@@ -57,6 +58,15 @@ def addons_config():
                     for glob in addons:
                         logging.debug("Expanding glob %s", glob)
                         for addon in iglob(os.path.join(SRC_DIR, repo, glob)):
+                            if not os.path.isdir(addon):
+                                continue
+                            addon_files = set(os.listdir(addon))
```
Comment (Yajo) [inline lib/odoobaselib/__init__.py]: I feel that `if not (os.path.isfile(os.path.join(addon, "__manifest__.py")) or os.path.isfile(os.path.join(addon, "__openerp__.py"))):` would be faster, don't you think? 🤔

Comment (PCatinean) [reply lib/odoobaselib/__init__.py]: Indeed, right you are.

File: lib/odoobaselib/__init__.py (hunk 2 above)
Comment (PCatinean) [reply lib/odoobaselib/__init__.py]: Indeed, I just thought that both conditions should be satisfied just like a regular Odoo module, though manifest should be ok. I did however check for both forms openerp and manifest so it keeps compatibility with V8.

File: lib/odoobaselib/__init__.py [hunk 4]
```diff
@@ -58,14 +58,13 @@ def addons_config():
...
-                            addon_files = set(os.listdir(addon))
-                            has_manifest = manifest_files & addon_files
-                            py_module = '__init__.py' in addon_files
-                            if not py_module or not has_manifest:
-                                logging.debug("Skipping '%s' as it is not a "
-                                    "valid Odoo module" % addon)
+                            manifests = [
+                                os.path.join(addon, m) for m in manifest_files
+                            ]
+                            if not any([os.path.isfile(m) for m in manifests]):
```
Comment (Yajo) [inline lib/odoobaselib/__init__.py]: Remove the brackets please.

File: lib/odoobaselib/__init__.py [hunk 5]
```diff
@@ -58,14 +58,13 @@ def addons_config():
...
+                                logging.debug(
+                                    "Skipping '%s' as it is not a valid Odoo "
+                                    "module" % addon)
```
Comment (Yajo) [inline lib/odoobaselib/__init__.py]: Replace `%` by `,` (a.k.a. `addon` should be function's 2nd argument).

File: lib/odoobaselib/__init__.py [hunk 6]
```diff
@@ -58,14 +58,13 @@ def addons_config():
...
-                                continue
-                            addon_files = set(os.listdir(addon))
-                            has_manifest = manifest_files & addon_files
-                            py_module = '__init__.py' in addon_files
-                            if not py_module or not has_manifest:
-                                logging.debug("Skipping '%s' as it is not a "
-                                    "valid Odoo module" % addon)
+                            manifests = [
+                                os.path.join(addon, m) for m in manifest_files
+                            ]
```
Comment (Yajo) [inline lib/odoobaselib/__init__.py]: Replace `[...]` for `(...)`, to save some iterations.

File: lib/odoobaselib/__init__.py [hunk 7]
```diff
@@ -57,6 +58,14 @@ def addons_config():
                     for glob in addons:
                         logging.debug("Expanding glob %s", glob)
                         for addon in iglob(os.path.join(SRC_DIR, repo, glob)):
+                            manifests = [
+                                os.path.join(addon, m) for m in manifest_files
+                            ]
+                            if not any([os.path.isfile(m) for m in manifests]):
```
Comment (Yajo) [inline lib/odoobaselib/__init__.py]: Remove brackets

File: lib/odoobaselib/__init__.py [hunk 8]
```diff
@@ -57,6 +58,14 @@ def addons_config():
                     for glob in addons:
                         logging.debug("Expanding glob %s", glob)
                         for addon in iglob(os.path.join(SRC_DIR, repo, glob)):
+                            manifests = [
+                                os.path.join(addon, m) for m in manifest_files
+                            ]
```
Comment (Yajo) [inline lib/odoobaselib/__init__.py]: Replace brackets for parenthesis

**Comment by PCatinean** [comment]
@Yajo not a problem, I get to learn more in the process and it should be just how you want it since it's your work after all. So never be afraid to keep insisting on my PR's I won't ever mind :smiley:

**Pull request merged**